### PR TITLE
Fixed PIT joins

### PR DESCRIPTION
**Pull request by obause**
Fixed an issue in joins when satellites without a load end date were used to build a PIT table

### CODE REVIEW

**Comment by tkirschke** [comment]
BigQuery Testing is passed.

**Review by tkirschke — COMMENTED** [review]
Have successfully tested it on BigQuery, requested Exasol testing from @bschlottfeldt . After passing Exasol testing, can be approved and merged.

**Comment by bschlottfeldt** [comment]
Please use the variable alias, not the hardcoded ledts. Here in Intersport we use a different naming convention for the LOAD END DATE TIMESTAMP, so this will not work when trying to build a PIT with a s0 satellite here.

File: macros/tables/exasol/pit.sql [hunk 1]
```diff
@@ -68,14 +69,20 @@ pit_records AS (
...
         {% for satellite in sat_names %}
         {%- set sat_columns = datavault4dbt.source_columns(ref(satellite)) %}
+        {%- if ledts|string|lower in sat_columns|map('lower') %}
         LEFT JOIN {{ ref(satellite) }}
+        {%- else %}
+        LEFT JOIN (
+            SELECT
+                {{ hashkey }},
+                {{ ldts }},
+                COALESCE(LEAD(ADD_SECONDS({{ ldts }}, -0.001)) OVER (PARTITION BY {{ hashkey }} ORDER BY {{ ldts }}),{{ datavault4dbt.string_to_timestamp( timestamp_format , end_of_all_times) }}) AS ledts
```
Comment (bschlottfeldt) [inline macros/tables/exasol/pit.sql:79]: use {{ ledts }}

File: macros/tables/snowflake/pit.sql [hunk 2]
```diff
@@ -68,14 +69,20 @@ pit_records AS (
...
         {% for satellite in sat_names %}
         {%- set sat_columns = datavault4dbt.source_columns(ref(satellite)) %}
+        {%- if ledts|string|lower in sat_columns|map('lower') %}
         LEFT JOIN {{ ref(satellite) }}
+        {%- else %}
+        LEFT JOIN (
+            SELECT
+                {{ hashkey }},
+                {{ ldts }},
+                COALESCE(LEAD({{ ldts }} - INTERVAL '1 MICROSECOND') OVER (PARTITION BY {{ hashkey }} ORDER BY {{ ldts }}),{{ datavault4dbt.string_to_timestamp(timestamp_format, end_of_all_times) }}) AS ledts
```
Comment (bschlottfeldt) [inline macros/tables/snowflake/pit.sql:79]: use {{ ledts }}

File: macros/tables/bigquery/pit.sql [hunk 3]
```diff
@@ -66,14 +67,20 @@ pit_records AS (
...
         {% for satellite in sat_names %}
         {%- set sat_columns = datavault4dbt.source_columns(ref(satellite)) %}
+        {%- if ledts|string|lower in sat_columns|map('lower') %}
         LEFT JOIN {{ ref(satellite) }}
+        {%- else %}
+        LEFT JOIN (
+            SELECT
+                {{ hashkey }},
+                {{ ldts }},
+                COALESCE(LEAD(TIMESTAMP_SUB({{ ldts }}, INTERVAL 1 MICROSECOND)) OVER (PARTITION BY {{ hashkey }} ORDER BY {{ ldts }}),{{ datavault4dbt.string_to_timestamp(timestamp_format, end_of_all_times) }}) AS ledts
```
Comment (bschlottfeldt) [inline macros/tables/bigquery/pit.sql:77]: use {{ ledts }}

Comment (tkirschke) [reply macros/tables/bigquery/pit.sql]: Hi @bschlottfeldt , your requested changes are now committed in this branch. If it works for you, you could approve the Review.

**Pull request merged**